src/consumption/elevate.js

```diff
@@ -0,0 +1,1827 @@
+		TASKS: 'projectTemplateTasks',
+		USER_ROLES: 'userRoles',
+		PROGRAMS: 'programs',
+		SOLUTIONS: 'solutions',
+		CERTIFICATE_TEMPLATE: 'certificateTemplates',
+		CERTIFICATE_BASE_TEMPLATE: 'certificateBaseTemplates',
+	})
+)
+
+/**
+ * To connect with the mongoDB with the given url
+ * @name connectMongo
+ * @param {Object} url - mongo url as returned from config
+ * @returns {Object} - Connection Object
+ */
+const connectMongo = async (url) => {
+	mongoConnection = new MongoDBConnection(url)
+	await mongoConnection.connect() // This returns the db, but it's not stored
+	// Get the database instance
+	return mongoConnection.getDb()
+}
+
+/**
+ * Format Project Template
+ * @name formatTemplate
+ * @param {Object} templateData - Project template data
+ * @returns {Object} - Response contains formatted template
+ */
+const formatTemplate = (templateData) => {
+	try {
+		let template = {
+			title: templateData.title,
+			tenantId: templateData.tenant_code,
+			description: templateData.objective || '',
+			keywords: utils.formatKeywords(templateData.keywords),
+			isDeleted: false,
+			createdBy: templateData.user_id,
+			updatedBy: templateData.user_id,
+			learningResources: utils.convertResources(templateData.learning_resources || []),
+			isReusable: true,
+			deleted: false,
+			status: common.PUBLISHED_STATUS,
+			externalId: utils.generateExternalId(templateData.title),
+			entityType: templateData?.entityType || '',
+			metaInformation: utils.formatProjectMetaInformation(templateData),
+			recommendedFor: [], //Initially empty
+			categories: [], //Initially empty
+			tasks: [], // Initially empty
+			taskSequence: [], // Initially empty
+			createdAt: new Date(),
+			updatedAt: new Date(),
+		}
+		return { success: true, template }
+	} catch (error) {
+		console.error('Error in formatTemplate:', error.message)
+		return { success: false, error: error.message }
+	}
+}
+
+/**
+ * Create and Find Categories
+ * @name processCategories
+ * @param {Object} categories - Categories Data
+ * @returns {Object} - Response contains categories data
+ */
+async function processCategories(categories, orgCode, tenantCode) {
+	try {
+		// Fetch a specific collection
+		const categoriesCollection = projectsMongoConnection.collection(COLLECTIONS_MAP.get('CATEGORIES'))
+
+		// Format categories
+		const formattedCategories = categories.map((category) => {
+			if (!category.label || !category.value) {
+				throw new Error('EACH_CATEGORY_MUST_BE_LABEL_AND_VALUE')
+			}
+			return {
+				label: category.label,
+				value: category.value,
+				formattedName: utils.formatToTitleCase(category.value),
+				externalId: category.value.replace(/_/g, '').toLowerCase(),
+			}
+		})
+
+		// Fetch existing categories by externalId
+		let existingCategories = []
+		const externalIds = formattedCategories.map((cat) => cat.externalId)
+		if (externalIds.length > 0) {
+			existingCategories = await categoriesCollection.find({ externalId: { $in: externalIds } }).toArray()
```

Comment:
@adithyadinesh0412 Can we fetch the categories including the tenant details as well